modules/fluid_properties/src/userobjects/TabulatedFluidProperties.C

```diff
@@ -491,6 +740,43 @@ TabulatedFluidProperties::s_from_p_T(Real p, Real T, Real & s, Real & ds_dp, Rea
   SinglePhaseFluidProperties::s_from_p_T(p, T, s, ds_dp, ds_dT);
 }
 
+Real
+TabulatedFluidProperties::g_from_v_e(Real v, Real e) const
+{
+  const Real p = p_from_v_e(v, e);
+  const Real T = T_from_v_e(v, e);
+  const Real s = s_from_p_T(p, T);
+  const Real h = h_from_p_T(p, T);
+  // this is simple the definition of Gibbs free energy
+  return h - T * s;
+}
+
+Real
+TabulatedFluidProperties::e_from_v_h(Real v, Real h) const
+{
+  if (!_construct_pT_from_ve)
+    mooseError("You must construct pT from ve tables when calling e_from_v_h.");
```

Comment:
maybe use `(p, T)` and `(v, e)` instead of `pT` and `ve`. Or `pt-` and `ve-tables`. It may not be obvious to a user what you are trying to tell them. Also mention, that you want them to use the `construct_pT_from_ve` param set to `true`. Otherwise this error message is fairly useless.